icu4c/source/i18n/plurrule.cpp

```diff
@@ -1732,6 +1752,23 @@ int32_t FixedDecimal::getVisibleFractionDigitCount() const {
     return visibleDecimalDigitCount;
 }
 
+bool FixedDecimal::operator==(const FixedDecimal &other) const {
+    return source == other.source && visibleDecimalDigitCount == other.visibleDecimalDigitCount
+        && decimalDigits == other.decimalDigits && exponent == other.exponent;
+}
+
+UnicodeString FixedDecimal::toString() const {
+    char pattern[15];
+    char buffer[20];
+    if (exponent == 0) {
+        sprintf(pattern, "%.%df", visibleDecimalDigitCount);
```

Comment:
Avoid sprintf in library code.  It is unsafe.  I'm also not sure if we can use it in library code.  Consider snprintf, but first, please check to see if there are any instances of snprintf in library code.  If not, then just format the number manually.

Reply:
Done.